### DISCUSSION TITLE
Lookup - get item.TextField

### DISCUSSION THREAD

**Discussion by kilroyFR** [serenity-is/Serenity#5824] [Help] [answered]
hey there, i made a custom lookup displaying 2 fields (mainly the id and text field) in getItemText.

Is it possible to get the idField and textField from a lookup for the current selection ? (i would like to avoid parsing the displayed text).

Thanks

**Comment by minhhungit** [comment]
lookup can return more fields not only Id and Text

![image](https://user-images.githubusercontent.com/2279508/123337182-09f4f000-d571-11eb-8c00-a4340dd6c755.png)

![image](https://user-images.githubusercontent.com/2279508/123337143-fe092e00-d570-11eb-8c54-92e19f96338a.png)

**Comment by kilroyFR** [comment, marked as answer]
sorry my bad, i would like to get the content of getItemText() defined in my LookupEditor (from my xxxDialog.ts)

EDIT :

got it.
this.form.MyLookupEditor.get_text()

**Reply by kilroyFR** [reply]
It works but i am getting a warning:

"Property 'get_text' is protected and only accessible within class 'Select2Editor<TOptions, TItem>' and its subclasses"

So i finally added ts-ignore token on previous line :
// @ts-ignore
this.form.MyLookupEditor.get_text()

**Reply by luongnv6** [reply]
use:
```this.form.MyLookupEditor.text```
or
```(this.form.MyLookupEditor as any).get_text()```
